### PR TITLE
Feat(eos_designs): Enable KERNELFIB_PROGRAM_ALL_ECMP for all wan routers

### PR DESCRIPTION
**Pull request by ayushmittal-arista**
## Change Summary
Enable KERNELFIB_PROGRAM_ALL_ECMP for all wan routers by setting environment variable for KernelFib.

## Related Issue(s)

Fixes #

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Add file agents.py in underlay python_modules to generate agent structured config for KernelFib
```
 {
     "name": "KernelFib", 
     "environment_variables": [
                {"name": "KERNELFIB_PROGRAM_ALL_ECMP", "value": "1"}
      ]
},
 ```

## How to test
`molecule`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/agents.py
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/agents.py:1]: What should be the copyright year here, @ClausHolbechArista / @guillaumeVilar

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/agents.py:1]: whatever the pre-commit inserts :)

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/agents.py:1]: 2024 IMO.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]
LGTM

**Pull request merged**